### ISSUE TITLE
Padding function may have a problem

### ISSUE THREAD

**Issue by qlwang25** [lancopku/bag-of-words#4]
https://github.com/lancopku/bag-of-words/blob/44eda4d0db292493a52fe0172804711e617c48f4/utils/data_helper.py#L86

```
def padding(data):
    src, tgt, original_src, original_tgt = zip(*data)

    src_len = [len(s) for s in src]
    src_pad = torch.zeros(len(src), max(src_len)).long()
    for i, s in enumerate(src):
        end = src_len[i]
        src_pad[i, :end] = torch.LongTensor(s[end-1::-1])

```

here, src_pad[i, :end] = torch.LongTensor(s[:end])
